ggml/src/ggml-cuda/ggml-cuda.cu

```diff
@@ -2825,6 +2826,21 @@ static bool ggml_cuda_can_fuse(const struct ggml_cgraph * cgraph, int node_idx,
     GGML_ASSERT(unary_ops.size() == num_unary);
 #endif
 
+    //special case for topk-moe
+    if (ops.size() == 5 && ops.begin()[0] == GGML_OP_SOFT_MAX && ops.begin()[1] == GGML_OP_RESHAPE && ops.begin()[2] == GGML_OP_ARGSORT
+        && ops.begin()[3] == GGML_OP_VIEW && ops.begin()[4] == GGML_OP_GET_ROWS) {
+
+        for (int i = 0; i < 5; i++) {
+            if (cgraph->nodes[node_idx + i]->op != ops.begin()[i]) return false;
+        }
+
```

Comment:
Isn't all this redundant since it is performed in `ggml_can_fuse`?

Reply:
I think the logic of this function should be that you always apply `ggml_can_fuse` first and only then do special-cases.